src/west/commands/command.py

```diff
@@ -67,13 +75,19 @@ def __init__(self, name, help, description, accepts_unknown_args=False):
                                      in its run() method. Otherwise, passing
                                      unknown arguments will cause
                                      UnknownArgumentsError to be raised.
+        :param requires_installation: if true, the command requires a
+                                      west installation to run. Running such a
+                                      command outside of any installation
+                                      (i.e. without a valid topdir) will exit
+                                      with an error.
         '''
         self.name = name
         self.help = help
         self.description = description
         self._accept_unknown = accepts_unknown_args
+        self.requires_installation = requires_installation
 
-    def run(self, args, unknown):
+    def run(self, args, unknown, topdir):
```

Comment:
Same here, what happens to existing commands that don't implement this? Will they break? If so, how do we handle users that update to the latest west but then want to bisect master?

Reply:
Nothing breaks because the do_run() signature is not changed incompatibly, and that's what extensions implement.

Reply:
Well, extensions and built-in commands - but built-ins don't matter, because they are built in and can be kept up to date. There just aren't any changes to make.